sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -59,14 +56,75 @@ case class ScalaUDF(
 
   override def toString: String = s"${udfName.getOrElse("UDF")}(${children.mkString(", ")})"
 
+  /**
+   * The analyzer should be aware of Scala primitive types so as to make the
+   * UDF return null if there is any null input value of these types. On the
+   * other hand, Java UDFs can only have boxed types, thus this parameter will
+   * always be all false.
```

Comment:
We need to make sure the comment is accurate. `Java UDFs can only have boxed types, thus this parameter will always be all false.` This is wrong now.

I agree that `Nil` is fine in this case, but the comment needs to be updated.